### PR TITLE
#3 - Update setting of variables version and tag  

### PR DESCRIPTION
**Pull request by mlee03**
In this PR,

changes have been made to the variables `tag` and `file_version`.   The variable `tag` has been removed.  Declaration of the variable `file_version` has been moved to a newly created `file_version.fh` file and is set to a default value of `unknown`.   With these changes, 

`call fms_write_version_number(version,tag)`

 in _subroutine coupler_init_ has been changed to 

`call fms_write_version_number('FULL_COUPLER_MOD', version)`

Note the change in the ordering of arguments, _i.e._ version now as the second argument, has been made to be consistent with modules in FMS.

Output from write_version_number in *logfile.*out is now changed from 
```
$Id$
$Name
```
to 
```
FULL_COUPLER_MOD
unknown
```

This PR depends on #104 and #106

### CODE REVIEW

File: full/full_coupler_mod.F90
Comment (bensonr) [inline full/full_coupler_mod.F90:136]: We no longer use tag in any meaningful way and you can delete these lines of code and any use of it later in the file.

File: full/full_coupler_mod.F90
Comment (bensonr) [inline full/full_coupler_mod.F90:130]: The convention has been to use:
`#ifdef _FILE_VERSION`
`  character(len=*), parameter :: version = _FILE_VERSION`
`#else`
`  character(len=*), parameter :: version = 'unknown'`
`#endif`

For now you may want to use a `version.fh` include to import this information.  In the future, we can come up with a method to populate the `version.fh` wtih proper information based on the build system and/or compiler in use.

Comment (mlee03) [reply full/full_coupler_mod.F90]: done.

**Review by thomas-robinson — DISMISSED** [review]

**Comment by thomas-robinson** [comment]
@rem1776 wait for 2a to be merged before merging this.

**Review by rem1776 — APPROVED** [review]

**Pull request merged**
